hathor/feature_activation/feature_service.py

```diff
@@ -0,0 +1,170 @@
+#  Copyright 2023 Hathor Labs
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#  http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+from typing import NamedTuple
+
+from hathor.feature_activation.feature import Feature
+from hathor.feature_activation.model.criteria import Criteria
+from hathor.feature_activation.model.feature_state import FeatureState
+from hathor.feature_activation.settings import Settings
+from hathor.transaction import Block
+
+
+class _Block(NamedTuple):
```

Comment:
My mistake, I didn't realize I could get it from metadata, I thought it was always a calculated attribute. I'll remove this class and create a `Block.height` property

Reply:
Maybe we should change `Block.calculate_height()` to `Block._calculate_height()`? (in a separate PR)

Reply:
Done in https://github.com/HathorNetwork/hathor-core/pull/613/commits/ff0735143eceb321f880c207bb6544b138ac58e8

Reply:
I think it's ok to rename it to `_calculate_height()`. We don't like using calculate attributes because they hide complexity and we feel that explicit is better than implicit. So I think we are better of with a `get_height()` method.

Reply:
Ahh, ok. Changed from `height` property to `get_height()` in https://github.com/HathorNetwork/hathor-core/commit/537e883413c022a020973b99b20ce2e08874460a

Reply:
Thanks!